src/gluonts/dataset/rolling_dataset.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from typing import Optional
+
+import numpy as np
+import pandas as pd
+from gluonts.dataset.common import Dataset
+from gluonts.dataset.util import to_pandas
+from pydantic import BaseModel
+
+
+class StepStrategy(BaseModel):
+    """
+    Removes datapoints equivalent to step_size for each iteration until
+    amount of data left is less than prediction_length
+
+    Parameters
+    ----------
+    prediction_length
+        The prediction length of the Predictor that the dataset will be
+        used with
+
+    Returns
+    -------
+    A partial function which yields the rolled windows
+    """
+
+    prediction_length: int
+    step_size: int = 1
+
+    def get_windows(self, window):
+        assert (
+            self.prediction_length > 0
+        ), """the step strategy requires a prediction_length > 0"""
+        assert self.step_size > 0, """step_size should be > 0"""
+
+        while len(window) >= self.prediction_length:
+            yield window
+            window = window[: -self.step_size]
+
+
+class NumSplitsStrategy(BaseModel):
+    prediction_length: int
+    num_splits: int
```

Comment:
I think it would be good to have a docstring similar to the one for the other strategy

Reply:
Done